### PR TITLE
feat(GridForm): drilled error states through GridForm to inputs

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## Drilled error states through GridForm to inputs

Previously, the `error` props were never set for `GridForm` text `Input`s, so even if their form validation failed, they'd still get a nice happy blue outline. No more!

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Screenshot of a form tet input with a blue outline
</td>
<td>
Screenshot of a form text input with a red outline
</td>
</tr>
</tbody>
</table>

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.9-alpha.eb51c0.0<br/>@codecademy/gamut-templates@2.5.8-alpha.eb51c0.0<br/>@codecademy/gamut@2.15.1-alpha.eb51c0.0<br/></pre>

File: packages/gamut/src/Form/styles/Select.module.scss [hunk 1]
```diff
@@ -6,6 +6,13 @@
   font-weight: normal;
 }
 
+.error {
+  border-color: $deprecated-swatches-red-400;
```
Comment (hmafzal) [inline packages/gamut/src/Form/styles/Select.module.scss:10]: Using deprecated colors?

File: packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormSelectInput/index.tsx [hunk 2]
```diff
@@ -6,13 +6,15 @@ import { GridFormSelectField } from '../../types';
 
 export type GridFormSelectInputProps = {
   className?: string;
+  error?: boolean;
```
Comment (hmafzal) [inline packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormSelectInput/index.tsx:9]: Where do we show the validation message? Or do we have a pattern for that yet?

File: packages/gamut/src/Form/styles/Select.module.scss (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Form/styles/Select.module.scss:10]: Heh, yeah, copy & paste from the text input... but I'll update to the closest non-deprecated one now. 😄

File: packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormSelectInput/index.tsx (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/gamut-templates/src/GridForm/GridFormInputGroup/GridFormSelectInput/index.tsx:9]: The rough chain for showing them is:

1. Reported to use as [`errors` from `useForm`](https://github.com/Codecademy/client-modules/blob/50b72e0b19383fddf842b3b3833cfac302dff837/packages/gamut-templates/src/GridForm/index.tsx#L24)
2. For each field, [retrieved under `errors[field.name]` as `errorMessage`](https://github.com/Codecademy/client-modules/blob/50b72e0b19383fddf842b3b3833cfac302dff837/packages/gamut-templates/src/GridForm/index.tsx#L44)
3. Rendered as a [`<FormError>` in `GridFormInputGroup`](https://github.com/Codecademy/client-modules/blob/eb51c04129bb364df33d5f240f9c6748bc37b05b/packages/gamut-templates/src/GridForm/GridFormInputGroup/index.tsx#L53)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.11-alpha.8ec156.0<br/>@codecademy/gamut-templates@2.6.3-alpha.8ec156.0<br/>@codecademy/gamut@2.17.1-alpha.8ec156.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e8ce276d55cb9037eb4c85b--codecademy-styleguide.netlify.com>https://5e8ce276d55cb9037eb4c85b--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e8ce276d55cb9037eb4c85b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@0.8.18-alpha.f3ad0c.0<br/>@codecademy/gamut@3.1.1-alpha.f3ad0c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5e9088844714bf85ed0ccc98--codecademy-styleguide.netlify.com>https://5e9088844714bf85ed0ccc98--codecademy-styleguide.netlify.com</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5e9088844714bf85ed0ccc98>Deploy Logs</a></p>

**Pull request merged**